test/e2e/e2e_local_source_upload_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/build/test/utils"
+)
+
+var _ = Describe("For a Kubernetes cluster with Tekton and build installed", func() {
```

Comment:
I agree! The complete test workflow will initially happen on the CLI repository (as part of [PR #86](https://github.com/shipwright-io/cli/pull/86)), since it depends on the client to utilize the local source upload feature end-to-end. As per:

https://github.com/otaviof/cli/blob/5c0cadc4c623920cafc43d1b77dbc481660129ad/test/e2e/upload.bats#L62-L64

Given the approach is consolidated on the CLI side, we can then reuse it here, if we conclude we want it here to.

What do you think about this approach, and, should we create a issue to keep track of it?

Reply:
Yes, an issue sounds fine.